### ISSUE TITLE
No elixir version specified. Version stays the same, does not work with 0.12.2

### ISSUE THREAD

**Issue by lau** [dynamo/dynamo#198]
The version seems to stay on 0.1.0-dev. The 0.12.2 tag also has version 0.1.0-dev.
see https://github.com/dynamo/dynamo/issues/195

Would it not be a good idea to have both the elixir version specified in the mix.exs?
And for the projects generated with mix.exs both the elixir version and the dynamo version should be specified in the mix.exs.

**Comment by lau** [comment]
I tried running tests on elixir master (0.12.4-dev) and dynamo master and got 4 errors running tests (see below).

Does it make sense to use travis to test the build if the elixir version is not specified? I get that it is a good idea to run on a cutting edge version of elixir. But how about specifying at least a commit id of elixir master, if the releases are too old?

I have never used Dynamo before and want to try it. But right now the recommended way of doing it does not seem to work.

```
1) test tests application code (Mix.TasksTest)
     ** (ExUnit.ExpectationError)
          expected: String.match?(output, sigil_r(<<"3 tests, 0 failures">>, []))
             to be: true
       instead got: false
     stacktrace:
       test/mix/tasks_test.exs:59: anonymous fn/0 in Mix.TasksTest.test tests application code/1
       (elixir) lib/file.ex:894: File.cd!/2
       test/mix/tasks_test.exs:45: Mix.TasksTest."test tests application code"/1

  2) test runs application code (Mix.TasksTest)
     ** (ExUnit.ExpectationError)
          expected: String.match?(output, sigil_r(<<"HelloRouter">>, []))
             to be: true
       instead got: false
     stacktrace:
       test/mix/tasks_test.exs:40: anonymous fn/0 in Mix.TasksTest.test runs application code/1
       (elixir) lib/file.ex:894: File.cd!/2
       test/mix/tasks_test.exs:36: Mix.TasksTest."test runs application code"/1

  3) test prints application filters (Mix.TasksTest)
     ** (ExUnit.ExpectationError)
          expected: String.match?(output, sigil_r(<<"filter Dynamo.Filters.Head">>, []))
             to be: true
       instead got: false
     stacktrace:
       test/mix/tasks_test.exs:29: anonymous fn/0 in Mix.TasksTest.test prints application filters/1
       (elixir) lib/file.ex:894: File.cd!/2
       test/mix/tasks_test.exs:25: Mix.TasksTest."test prints application filters"/1

  4) test compiles an application (Mix.TasksTest)
     ** (ExUnit.ExpectationError)
          expected: String.match?(output, sigil_r(<<"Compiled web/routers/application_router.ex">>, []))
             to be: true
       instead got: false
     stacktrace:
       test/mix/tasks_test.exs:10: anonymous fn/0 in Mix.TasksTest.test compiles an application/1
       (elixir) lib/file.ex:894: File.cd!/2
       test/mix/tasks_test.exs:6: Mix.TasksTest."test compiles an application"/1
```

**Comment by devinus** [comment]
@lau Those errors are unrelated, and I've fixed them. Elixir master changes everyday and we strive to keep Dynamo master working on Elixir master. I've tested the path using Dynamo tag `elixir-0.12.2` and it works fine. I suspect you're forgetting to redo `mix deps.get` and `mix compile` when you checkout the tag. And yes, the `elixir-0.12.2` tag works with Elixir 0.12.2 perfectly -- we use it in production every day.
